snowex_db/upload/layers.py

```diff
@@ -197,8 +304,70 @@ def _add_entry(self, session, row: pd.Series, metadata: ProfileMetaData):
             site=site,
             # Arguments from kwargs
             depth=row["depth"],
-            bottom_depth=row["bottom_depth"],
+            bottom_depth=row.get("bottom_depth"),
             comments=row["comments"],
             value=row["value"],
         )
         return new_entry
+
+
+class UploadProfileBatch(BatchBase):
+    """
+    Class for submitting multiple files of profile type data.
+
+    Attributes:
+        smp_log_f: CSV providing metadata for profile_filenames.
+    """
+
+    UploaderClass = UploadProfileData
+
+    def push(self):
+        """
+        An overwritten push function to account for managing SMP meta data.
+        """
+
+        self.start = time.time()
+
+        i = 0
+
+        # TODO: uncertain here
+        # if self.smp_log_f is not None:
+        #     self.smp_log = SMPMeasurementLog(self.smp_log_f)
+        # else:
+        #     self.smp_log = None
+
+        # Keep track of whether we are using a site details file for each profile
+        smp_file = False
+
+        # Read the data and organize it, remap the names
+        # if not isinstance(self.smp_log, type(None)):
+        #     self.log.info(
+        #         'Processing SMP profiles with SMP measurement log...')
+        #     smp_file = True
+        #     self.meta['header_sep'] = ':'
+
+        # Loop over all the ssa files and upload them
+        if self.n_files != -1:
+            self.filenames[0:self.n_files]
+
+        for i, f in enumerate(self.filenames):
+
+            # if smp_file:
+            #     extras = self.smp_log.get_metadata(f)
+            #     meta.update(extras)
```

Comment:
And again SMP stuff

Reply:
We already do some parsing of the SMP files before hand, we could reorganize them more to look like the other csv files and simplify this logic some. i.e. the measurements log file could be interpreted at the upload script level